auction-app-backend/src/test/java/com/atlantbh/auctionappbackend/controllerTest/AuthControllerTest.java

```diff
@@ -0,0 +1,69 @@
+package com.atlantbh.auctionappbackend.controllerTest;
+
+import com.atlantbh.auctionappbackend.api.ApiConfig;
+import com.atlantbh.auctionappbackend.controller.AuthController;
+import com.atlantbh.auctionappbackend.domain.User;
+import com.atlantbh.auctionappbackend.security.JwtConfig;
+import com.atlantbh.auctionappbackend.security.JwtUtil;
+import com.atlantbh.auctionappbackend.service.UserService;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.MediaType;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+
+import javax.servlet.Filter;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.when;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.content;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@WebMvcTest(AuthController.class)
+public class AuthControllerTest {
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    @MockBean
+    private UserService userService;
+    @MockBean
+    private JwtUtil jwtUtil;
+    @MockBean
+    private JwtConfig jwtConfig;
+    @MockBean
+    private PasswordEncoder passwordEncoder;
+    @MockBean
+    private ApiConfig apiConfig;
+    @MockBean(name = "springSecurityFilterChain")
+    private Filter springSecurityFilterChain;
+
+
+    @Test
+    public void greetingShouldReturnMessageFromService() throws Exception {
```

Comment:
The whole test class is a little problematic. At this moment this test fails as I'm not sure how to implement correct behavior. MockMvc always returns status code 200, and an empty response body (no matter which route I try, even if it isn't implemented at all). I suspect that the problem lies in how I used mocking. Because of this I created the AuthControllerNonMockTest class, which works fine, but doesn't use any mocking implying that the whole server starts when we run the tests. I suggest removing this class, and renaming the NonMock class, if we can make the tests use the H2 embedded db somehow.